Src/AutoFixture/Kernel/FieldSpecification.cs

```diff
@@ -95,7 +95,9 @@ private bool FieldIsCompatibleWithTargetType(object request)
 
         private bool FieldHasTargetName(object request)
         {
-            return ((FieldInfo)request).Name == this.targetName;
+            return ((FieldInfo)request)
+                   .Name
+                   .Equals(this.targetName, StringComparison.OrdinalIgnoreCase);
```

Comment:
> Isn't this a breaking change?

I was on the fence about making this change. My initial thought was to introduce a flag in the constructor that would determine whether the comparison should be CS or CI. In order to avoid breaking the class' public API, that flag would have to be an _optional_ parameter with a default value that indicated a CS comparison.

After giving it some more thought, however, I came to the conclusion that making the comparison CI by default wouldn't detract any previous matches but rather expand them. The resulting set would either be equal or a proper superset of the previous one. Is it a breaking change, then?

Well, the new behavior _could_ potentially be a problem if someone were to use AutoFixture to create an object model where multiple members of the same kind (i.e. properties, fields or parameters) have different semantic values based on the case of their names. For example:

```csharp
public class Foo
{
    public string Bar;
    public string BAR;
}
```

Assuming this model would even have a practical application, if the case is the only difference aren't `Bar` and `BAR` really the same thing?

So, to answer the question, this might be considered a breaking change _conceptually_, since it does change the existing behavior by including additional results, but I don't think it is a breaking change in practice.

Having said that, I realize we're still in the land of assumptions, so I'm going to go ahead and make the CI comparison optional like I meant to do initially. That would also make the specifications more flexible, as you pointed out.